src/evaluators/ClassDeclaration.js

```diff
@@ -10,11 +10,272 @@
 /* @flow */
 
 import type { Realm } from "../realm.js";
-import type { LexicalEnvironment } from "../environment.js";
+import type { LexicalEnvironment, Reference } from "../environment.js";
 import type { Value } from "../values/index.js";
-import type { Reference } from "../environment.js";
-import type { BabelNodeClassDeclaration } from "babel-types";
+import { NullValue, EmptyValue, ObjectValue } from '../values/index.js';
+import type { BabelNodeClassDeclaration, BabelNodeExpression, BabelNodeClassMethod } from "babel-types";
+import parse from "../utils/parse.js";
+import {
+  HasOwnProperty,
+  SetFunctionName,
+  IsConstructor,
+  Get,
+  MakeConstructor,
+  CreateMethodProperty,
+  DefineMethod,
+  NewDeclarativeEnvironment,
+  MakeClassConstructor,
+  ObjectCreate,
+  ConstructorMethod,
+  GetValue,
+  IsStatic,
+  PropertyDefinitionEvaluation,
+  InitializeBoundName,
+  NonConstructorMethodDefinitions,
+} from "../methods/index.js";
+import { AbruptCompletion } from "../completions.js";
+import invariant from "../invariant.js";
 
+function EvaluateClassHeritage(realm: Realm, ClassHeritage: BabelNodeExpression, strictCode: boolean): ObjectValue | null {
+  let ref = realm.getRunningContext().lexicalEnvironment.evaluate(ClassHeritage, strictCode);
+  let val = GetValue(realm, ref);
+  if (!(val instanceof ObjectValue)) {
+    return null;
+  }
+  return val;
+}
+
+function ClassDefinitionEvaluation(realm: Realm, ast: BabelNodeClassDeclaration, className: string | void, strictCode: boolean, env: LexicalEnvironment) {
+  // 1. Let lex be the LexicalEnvironment of the running execution context.
```

Comment:
Which section of the ecma spec do these comments belong to? Please add high-level comment like "// ECMA262 8.1.1".